collector/timex.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !notimex
+
+package collector
+
+// #include <sys/timex.h>
+import "C"
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// The system clock is not synchronized to a reliable server.
+	timeError = C.TIME_ERROR
+)
+
+type timexCollector struct {
+	offset, freq, syncStatus typedDesc
+}
+
+func init() {
+	Factories["timex"] = NewTimexCollector
+}
+
+// NewTimexCollector returns a new Collector exposing adjtime(3) stats.
+func NewTimexCollector() (Collector, error) {
+	const subsystem = "timex"
+
+	return &timexCollector{
+		offset: typedDesc{prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "offset_seconds"),
+			"Time offset in between local system and reference clock.",
+			nil, nil,
+		), prometheus.GaugeValue},
+		freq: typedDesc{prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "freq"),
+			"Local clock frequency adjustment.",
+			nil, nil,
+		), prometheus.GaugeValue},
```

Comment:
As far as I understand, `freq` is exported to understand if anything is going wrong with local clock stability.
Do you consider exporting `maxerror` useful for alike purpose? As far as I see, it shows ~age of the last adjustment sent to Linux kernel initialized with some initial value (systemd-timeyncd uses 0, chrony uses `offset + root_delay / 2 + root_disp`).